src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/SpecialDirectories.vb

```diff
@@ -0,0 +1,156 @@
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Return the directory that contain temporary files for the current user.
+        ''' </summary>
+        ''' <value>A String containing the path to the temporary directory for the current user.</value>
+        ''' <remarks>
+        ''' According to Win32 API document, GetTempPath should always return a value even if TEMP and TMP = "".
+        ''' Also, this is not updated if TEMP or TMP is changed in Windows. The reason is
+        '''     each process has its own copy of the environment variables and this copy is not updated.
+        ''' </remarks>
+        Public Shared ReadOnly Property Temp() As String
+            Get
+                Return GetDirectoryPath(IO.Path.GetTempPath(), SR.IO_SpecialDirectory_Temp)
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Returns the directory that serves as a common repository for data files
+        ''' from your application used only by the current user.
+        ''' </summary>
+        ''' <value>A String containing the path to the directory your application can use to store data for the current user.</value>
+        ''' <remarks>
+        ''' We chose to use UserAppDataPath instead of LocalUserAppDataPath since this directory
+        ''' will work with Roaming User as well.
+        ''' </remarks>
+        Public Shared ReadOnly Property CurrentUserApplicationData() As String
+            Get
+                Throw New PlatformNotSupportedException(GetResourceString(SR.IO_SpecialDirectoryNotExist, "ApplicationData"))
+                'Return GetDirectoryPath(System.Windows.Forms.Application.CommonAppDataPath, SR.IO_SpecialDirectory_AllUserAppData)
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Returns the directory that serves as a common repository for data files
+        ''' from your application used by all users.
+        ''' </summary>
+        ''' <value>A String containing the path to the directory your application can use to store data for all users.</value>
+        ''' <remarks>
+        ''' If a path does not exist, one is created in the following format
+        ''' C:\Documents and Settings\All Users\Application Data\[CompanyName]\[ProductName]\ProductVersion
+        ''' Either CompanyName or ProductName may be left out but not both
+        ''' Product version is required but will default to "0.0.0.0"
+        '''
+        ''' See above for reason why we don't use System.Environment.GetFolderPath(*).
+        ''' </remarks>
+        Public Shared ReadOnly Property AllUsersApplicationData() As String
+            Get
+                Throw New PlatformNotSupportedException(GetResourceString(SR.IO_SpecialDirectoryNotExist, "ApplicationData"))
+                'Return GetDirectoryPath(System.Windows.Forms.Application.UserAppDataPath, SR.IO_SpecialDirectory_UserAppData)
```

Comment:
The specific resource string seems misleading here. Please create `PlatformNotSupportedException()` with no arguments. And please remove the commented out code.

Reply:
It exists, I will use it.